xrtl/testing/render_test.h

```diff
@@ -0,0 +1,49 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef XRTL_TESTING_RENDER_TEST_H_
+#define XRTL_TESTING_RENDER_TEST_H_
+
+#include <string>
+
+#include "xrtl/testing/gtest.h"
+
+namespace xrtl {
+namespace testing {
+
+// Base for image pixel diffing tests.
+//
+// Usage:
+//  class MyTest : public testing::RenderTest {};
+//  TEST_F(MyTest, DoSomething) {
+//    auto pixel_data = ...;
+//    EXPECT_TRUE(CompareImages(path_to_golden, pixel_data, width, height));
+//  }
+class RenderTest : public ::testing::Test {
```

Comment:
It'd be useful if RenderTest actually dealt with rendering and pixel comparison was orthogonal. That way things that have nothing to do with rendering (like image decoding/encoding, texture compression, etc) can use the pixel comparison. For that, testing/pixel_diff.h or testing/image_comparison.h or something could contain these comparison functions and be included from any test without forcing that test to subclass RenderTest.